src/index.coffee

```diff
@@ -117,15 +118,33 @@ progenyConstructor = (mode, settings = {}) ->
 				if path in depsList
 					callback()
 				else
-					depsList.push path if potentialDeps
-					fs[mode].readFile path, encoding: 'utf8', (err, source) ->
-						return callback() if err
-						depsList.push path unless potentialDeps or path in depsList
-						parseDeps path, source, depsList, callback
+					if glob.hasMagic path
+						glob path, (err, files) ->
+							return callback() if err
+							each files, (path, callback) ->
+								depsList.push path if potentialDeps
+								addDep path, depsList, callback
+							, callback
+					else
+						depsList.push path if potentialDeps
+						ext = ".#{extension}"
+						modulePath = path.replace ext, "#{sysPath.sep}index#{ext}"
+						fs.exists modulePath, (exists) ->
```

Comment:
The `fs.exists` call is redundant. Just add any path you want to check to the list, and if it doesn't exist when `readFile` happens it'll get pruned out.